passkey-authenticator/src/user_validation.rs

```diff
@@ -1,11 +1,26 @@
+use passkey_types::{ctap2::Ctap2Error, Passkey};
+
 #[cfg(doc)]
 use crate::Authenticator;
 
+/// The result of a user validation check.
+#[derive(Clone, Copy, PartialEq)]
+pub struct UserCheck {
+    /// Indicates whether the user was present.
+    pub presence: bool,
+
+    /// Indicates whether the user was verified.
+    pub verification: bool,
+}
+
 /// Pluggable trait for the [`Authenticator`] to do user interaction and verification.
-#[async_trait::async_trait]
-#[cfg_attr(any(test, feature = "testable"), mockall::automock)]
+#[cfg_attr(any(test, feature = "testable"), mockall::automock(type PasskeyItem = Passkey;))]
 #[cfg_attr(any(test, feature = "testable"), allow(clippy::unused_async))] // Generated by the `mockall` macro.
+#[async_trait::async_trait]
```

Comment:
Changing the order to these seems to be the cause of some of the required test changes that remove the `Box::pin(async {})`, seems like by putting the async_trait macro at the bottom it's not applying to the mocks anymore. 

The tests will work either way, but I think it would be better to move the async_trait macro to the top, to better reflect what the real code will do, which also would mean less changes needed to the tests.

Reply:
Oh I actually made this change intentionally since it's recommended by the mockall crate (see the message of the first commit in this PR)

Reply:
Oh that's interesting, yeah makes sense following their docs then. This should be an easy change to upstream too